subprojects/core/src/main/java/org/gradle/api/tasks/GradleBuild.java

```diff
@@ -89,48 +88,6 @@ public void setDir(Object dir) {
         getStartParameter().setCurrentDir(getProject().file(dir));
     }
 
-    /**
-     * Returns the build file that should be used for this build. Defaults to {@value
-     * org.gradle.api.Project#DEFAULT_BUILD_FILE} in the project directory.
-     *
-     * @return The build file. May be null.
-     * @deprecated Use {@link #getDir()} instead to get the root of the nested build.
-     * This method will be removed in Gradle 8.0.
-     */
-    @Nullable
-    @Optional
-    @PathSensitive(PathSensitivity.NAME_ONLY)
-    @InputFile
-    @Deprecated
-    public File getBuildFile() {
-        return getStartParameter().getBuildFile();
-    }
-
-    /**
-     * Sets the build file that should be used for this build.
-     *
-     * @param file The build file. May be null to use the default build file for the build.
-     * @since 4.0
-     * @deprecated Use {@link #setDir(File)} instead to set the root of the nested build.
-     * This method will be removed in Gradle 8.0.
-     */
-    @Deprecated
-    public void setBuildFile(@Nullable File file) {
-        setBuildFile((Object) file);
-    }
-
-    /**
-     * Sets the build file that should be used for this build.
-     *
-     * @param file The build file. May be null to use the default build file for the build.
-     * @deprecated Use {@link #setDir(Object)} instead to set the root of the nested build.
-     * This method will be removed in Gradle 8.0.
-     */
-    @Deprecated
-    public void setBuildFile(@Nullable Object file) {
```

Comment:
We may not be able to remove this right now. From looking at the test failures, we have a number of tests that leverage that feature for running.
You could investigate how to get rid of them, but it might be a much bigger effort than initially thought.

Reply:
Thanks @ljacomet I'm trying to fix the failing tests but there are a number of them, and it doesn't look straight forward indeed.

Are there alternatives other than removing it, e.g. adding a deprecation warning? I don't see how could make sense other than removing the deprecated methods and fixing the tests.